Design.md

```diff
@@ -0,0 +1,160 @@
+# Job Worker Design Doc
+
+Design doc for a prototype job worker service that provides an API to run arbitrary Linux processes.
+
+## Level 2 requirements
+
+- Library: Worker library with methods to start/stop/query status and get the output of a job.
+- API:
+    - HTTPS API to start/stop/get status of a running process.
+    - Use mTLS authentication and verify client certificate. Set up strong set of cipher suites for TLS and good crypto
+      setup for certificates. Do not use any other authentication protocols on top of mTLS.
+    - Use a simple authorization scheme.
+- CLI:
+    - CLI should be able to connect to worker service and schedule several jobs.
+    - CLI should be able to query result of the job execution and fetch its output.
+
+## Job object
+
+- `ID string` unique identifier of a job
+- `Command string` Linux command to be executed
+- `Arguments []string` list of arguments for the command
+- `Status string` current status of the job
+- `OutputBuffer bytes.Buffer` links stderr and stdout of the job process, it will be hidden from API response
```

Comment:
nit: you will need some synchronization around this, if you want to forward both stdout and stderr to the same buffer